ext/TrixiConvexECOSExt.jl

```diff
@@ -0,0 +1,141 @@
+# Package extension for adding Convex-based features to Trixi.jl
+module TrixiConvexECOSExt
+
+# Required for coefficient optimization in P-ERK scheme integrators
+if isdefined(Base, :get_extension)
+    using Convex
+    using ECOS
+else
+    # Until Julia v1.9 is the minimum required version for Trixi.jl, we still support Requires.jl
+    using ..Convex
+    using ..ECOS
+end
+
+# Use other necessary libraries
+using LinearAlgebra: eigvals
+const MOI = Convex.MOI
+
+# Use additional symbols that are not exported
+using Trixi: PolynomialOptimizer, @muladd
+
+# Import functions such that they can be extended with new methods
+import .PolynomialOptimizer: filter_eigvals, bisection, undo_normalization!
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+function stability_polynomials(cons_order, num_stage_evals, num_eig_vals,
+                               normalized_powered_eigvals_scaled, pnoms,
+                               gamma::Variable)
+    # Initialize with zero'th order (z^0) coefficient
+    for i in 1:num_eig_vals
+        pnoms[i] = 1.0
+    end
+
+    # First `cons_order` terms of the exponential
+    for k in 1:cons_order
+        for i in 1:num_eig_vals
+            pnoms[i] += normalized_powered_eigvals_scaled[i, k]
+        end
+    end
+
+    # Contribution from free coefficients
+    for k in (cons_order + 1):num_stage_evals
+        pnoms += gamma[k - cons_order] * normalized_powered_eigvals_scaled[:, k]
+    end
+
+    # For optimization only the maximum is relevant
+    return maximum(abs(pnoms))
+end
+
+"""
+bisection()
+
+The following structures and methods provide a simplified implementation to 
+discover optimal stability polynomial for a given set of `eig_vals`
+These are designed for the one-step (i.e., Runge-Kutta methods) integration of initial value ordinary 
+and partial differential equations.
+
+- Ketcheson and Ahmadia (2012).
+Optimal stability polynomials for numerical integration of initial value problems
+[DOI: 10.2140/camcos.2012.7.247](https://doi.org/10.2140/camcos.2012.7.247)
+"""
+function bisection(cons_order, num_eig_vals, num_stage_evals, dtmax, dteps, eig_vals)
```

Comment:
I think `bisection` is not an optimal name:
* It is very generic and thus suggests it can be used for any kind of bisection (which is clearly not the intent here)
* A noun implies that this is what the function returns (like a constructor) - if you want to indicate that this is the action the function performs, it should rather be something like `bisect`